src/resources.js

```diff
@@ -0,0 +1,23 @@
+import xdr from './xdr';
+
+export function newSorobanData() {
+  return sorobanDataFromFootprint([], []);
+}
+
+export function sorobanDataFromFootprint(readOnly, readWrite) {
```

Comment:
so for my use case I think I needed to keep the rest of the resources, so I set it like this 
```
const sorobanTxData = xdr.SorobanTransactionData.fromXDR(
    simulation.transactionData,
    "base64",
  );
  sorobanTxData.resources().footprint(footprint);
  txnBuilder.setSorobanData(sorobanTxData);
```

so basically I merged a new simulation's resources with an old footprint, would I still be able to do that if `sorobanDataFromFootprint` 0s out the other keys of the SorobanResources?

Reply:
yeah so you wouldn't want to use these. instead, you'd want this:

```js
txnBuilder
  .setSorobanData(sorobanTxData)
  .setFootprint(footprint.readOnly(), footprint.readWrite());
```
for this: the latter call will leave the existing fields alone.

Reply:
After the refactor based on @sreuland's comment, this would now be:

```js
txnBuilder.setSorobanData(
  new SorobanDataBuilder(sorobanTxData)
    .setReadOnly(footprint.readOnly(), 
    .setReadWrite(footprint.readWrite())
    .build()
);